# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1319734766 25200
#      Thu Oct 27 09:59:26 2011 -0700
# Node ID fc9f34e174864d8ad93733f65ccb4a3cf0dd1402
# Parent  227130df89eb3f7fa77af06891fecfdfe5fc6844
display.m: Extend display to work on builtin classes (Bug #34637)

* display.m: Extend display to work on builtin classes.

diff --git a/scripts/general/display.m b/scripts/general/display.m
--- a/scripts/general/display.m
+++ b/scripts/general/display.m
@@ -28,14 +28,23 @@
 ## @noindent
 ## where Octave is required to display the contents of a variable of the
 ## type "myclass".
 ##
 ## @seealso{class, subsref, subsasgn}
 ## @end deftypefn
 
 function idx = display (a)
-  if (nargin == 1)
-    error ("display: not defined for class \"%s\"", class(a));
-  else
+
+  if (nargin != 1)
     print_usage ();
   endif
+ 
+  ## Only reason we got here is that there was no overloaded display()
+  ## function for object a.  This may mean it is a built-in.
+  str = disp (a);
+  if (isempty (strfind (str, "<class ")))
+    disp (str);   
+  else
+    error ('display: not defined for class "%s"', class (a));
+  endif
+
 endfunction
